xarray/tests/test_dataarray.py

```diff
@@ -4079,6 +4080,49 @@ def test_rank(self):
         y = DataArray([0.75, 0.25, np.nan, 0.5, 1.0], dims=("z",))
         assert_equal(y.rank("z", pct=True), y)
 
+    # TODO: use fixture or randomly created array instead of load_dataset
+    @requires_scipy_or_netCDF4
+    def test_corr(self):
+        # self: Load demo data and trim it's size
+        ds = xr.tutorial.load_dataset("air_temperature")
+        air = ds.air[:18, ...]
+        # other: select missaligned data, and smooth it to dampen the correlation with self.
+        air_smooth = (
+            ds.air[2:20, ...].rolling(time=3, center=True).mean(dim="time")
+        )  # .
+        # A handy function to select an example grid
+
+        def select_pts(da):
+            return da.sel(lat=45, lon=250)
+
+        # Test #1: Misaligned 1-D dataarrays with missing values
+        ts1 = select_pts(air.copy())
+        ts2 = select_pts(air_smooth.copy())
+
+        def pd_corr(ts1, ts2):
+            """Ensure the ts are aligned and missing values ignored"""
+            # ts1,ts2 = xr.align(ts1,ts2)
+            valid_values = ts1.notnull() & ts2.notnull()
+
+            ts1 = ts1.where(valid_values, drop=True)
+            ts2 = ts2.where(valid_values, drop=True)
+
+            return ts1.to_series().corr(ts2.to_series())
+
+        expected = pd_corr(ts1, ts2)
+        actual = ts1.corr(ts2)
+        np.allclose(expected, actual)
```

Comment:
You can use xarray's `assert_allclose` and you get some additional checks for free

Reply:
I think this (or using `assert np.allclose(...)`) actually *has* to be done, `np.allclose` only compares its arguments and returns a boolean: the tests would still pass even if `expected` and `actual` have different values.

Reply:
So, I mark this as resolved, please correct me if I am wrong!

Reply:
I don't think this is resolved yet: `np.allclose` is a function that compares its inputs, in this case `expected` and `actual`  and returns a boolean (like `==`, but using a threshold to compensate numerical errors). As we don't do anything with that boolean, it goes out of scope and the tests won't be able to detect bugs in the code they should be testing.

So the minimal fix I offered as an alternative is
```suggestion
        assert np.allclose(expected, actual)
```
which would raise an `AssertionError` if `np.allclose` returns `False`, allowing `pytest` to detect that the test failed. This works, but I think in this case is not enough: `actual` is a `DataArray` (which is consistent with other `xarray` operations) whereas `expected` is a `np.float64`.

What @max-sixty suggested would be
```suggestion
        assert_allclose(expected, actual)
```
`assert_allclose` expects `expected` and `actual` to be the same type of `xarray` objects (i.e. `DataArray` or `Dataset` objects) and compares like `np.allclose`, but raises an `AssertionError` if the comparison would return `False` (along with other checks and providing a nice error report). That means that the suggested change won't be enough, we need to wrap the result of `pandas.Series.corr` (a `np.float64` instance) into a `DataArray` while making sure it has the correct coords.